compiler+runtime/src/cpp/jank/runtime/obj/uuid.cpp

```diff
@@ -0,0 +1,67 @@
+#include <boost/functional/hash.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <jank/runtime/obj/uuid.hpp>
+#include <jank/runtime/rtti.hpp>
+
+namespace jank::runtime::obj
+{
+  uuid::uuid()
+  {
+    boost::uuids::random_generator gen;
+    value = gen();
+  }
+
+  uuid::uuid(native_persistent_string const &s)
+  {
+    value = boost::lexical_cast<boost::uuids::uuid>(s);
+  }
+
+  native_bool uuid::equal(object const &o) const
+  {
+    if(o.type != object_type::uuid)
+    {
+      return false;
+    }
+
+    auto const s(expect_object<uuid>(&o));
+    return s->value == value;
+  }
+
+  static void to_string_impl(boost::uuids::uuid value, util::string_builder &buff)
```

Comment:
`boost::uuids::uuid` is a struct type, so please pass it by `const &` to avoid copying.